### ISSUE TITLE
[Feature] Implement KV Compressor of DeepSeek-V4

### ISSUE THREAD

**Issue by bumble0918** [Inspiron-st/pypto-lib#198]
### Summary

Implement the KV Compressor component for DeepSeek-V4 using PyPTO language, completing the `deepseek_v4_decode_compressor_draft.py` file based on the Hugging Face DeepSeek-V4-Pro model implementation.

### Motivation / Use Case

The DeepSeek-V4 model requires an efficient KV Compressor to handle its million-scale context length capabilities. This component is critical for:
1. Supporting incremental decoding with memory-efficient KV caching
2. Maintaining compatibility with the official DeepSeek-V4-Pro implementation
3. Enabling the use of DeepSeek-V4 in PyPTO-Lib for NPU-accelerated inference
4. Providing a reference implementation for attention KV compression techniques

The implementation is based on the Hugging Face DeepSeek-V4-Pro model, ensuring consistency with the state-of-the-art transformer architecture.

### Function Signature
```python
@pl.function(type=pl.FunctionType.Opaque)
def deepseek_v4_decode_compressor(
    self,
    x:               pl.Tensor[[B, S, D],                       pl.BF16],
    kv_state:        pl.InOut[pl.Tensor[[B, STATE_LEN, OUT_DIM], pl.FP32]],
    score_state:     pl.InOut[pl.Tensor[[B, STATE_LEN, OUT_DIM], pl.FP32]],
    wkv:             pl.Tensor[[OUT_DIM, D],                    pl.BF16],
    wgate:           pl.Tensor[[OUT_DIM, D],                    pl.BF16],
    ape:             pl.Tensor[[RATIO, OUT_DIM],                pl.FP32],
    weight:          pl.Tensor[[HEAD_DIM],                      pl.BF16],
    cos:             pl.Tensor[[1, ROPE_DIM],                   pl.BF16],
    sin:             pl.Tensor[[1, ROPE_DIM],                   pl.BF16],
    start_pos:       pl.Tensor[[B],                             pl.INT32],
    should_compress: pl.Tensor[[1],                             pl.INT32],
    out:             pl.Out[pl.Tensor[[B, HEAD_DIM],            pl.BF16]],
):
    # Implementation details
```

### Core Behavior

1. **Projection Phase:**
   - Compute KV and score projections from input tokens
   - Add positional embeddings for the current slot

2. **State Management:**
   - Update state buffers with new values for incremental decoding
   - Handle state shifting when compression occurs

3. **Compression Logic:**
   - Concatenate overlapping window halves when ratio=4
   - Apply softmax weighted sum for compression
   - Update state buffers after compression

4. **Post-Processing:**
   - Apply RMSNorm normalization
   - Implement RoPE (Rotary Position Embedding)
   - Cast to appropriate output type

### Key Parameters
- Batch size: B = 16
- Sequence length: S = 1
- Hidden dimension: D = 4096
- Head dimension: HEAD_DIM = 512
- RoPE dimension: ROPE_DIM = 64
- Compression ratio: RATIO = 4
- State length: STATE_LEN = 8
- Output dimension: OUT_DIM = 1024

### Alternatives Considered

1. **Direct PyTorch port without state management**
   - Rejected: Would not support incremental decoding

2. **Simplified compression strategy**
   - Rejected: Would not match the reference model's behavior and performance characteristics

3. **Different data type configurations**
   - Rejected: To maintain compatibility with the original model's precision requirements

### Source Reference
**Hugging Face Model Source:** https://huggingface.co/deepseek-ai/DeepSeek-V4-Pro/blob/main/inference/model.py

**Target File:** `examples/models/deepseek/v4/deepseek_v4_decode_compressor_draft.py`

### Testing Support

The implementation includes:
- A PyTorch golden function (`golden_deepseek_v4_decode_compressor`) for validation
- Comprehensive tensor specifications for input/output compatibility
- Multi-platform test support (a2a3, a2a3sim, a5, a5sim)

### Running Tests
```bash
python3 deepseek_v4_decode_compressor_draft.py --platform <platform> --device <device_id>
```

### Notes
This implementation specifically handles the decode incremental path with ratio=4, head_dim=512, rotate=False, and overlap=True, which is the main attention path for DeepSeek-V4.
